### ISSUE TITLE
Regression in indexing by vectors in large datasets

### ISSUE THREAD

**Issue by mbauman** [JuliaComputing/JuliaDB.jl#210]
Between 0.8.1 and 0.8.3 we lost the ability to index into the TrueFX dataset with a vector of currency pairs.  I've been trying to get a MWE but simply creating a smaller `ndsparse` doesn't seem to do the trick.  It also seems to be dependent upon where the chunks live.  Here's the issue:

```julia
julia> data = load("bindata/full-2016/")
2-d Distributed NDSparse with 1671681959 values (2 field named tuples) in 80 chunks:
pair       timestamp               │ bid     ask
───────────────────────────────────┼───────────────
"AUD/JPY"  2016-01-04T00:00:00.108 │ 87.641  87.651
…

julia> data[["AUD/JPY","EUR/JPY"],:]
2-d Distributed NDSparse of (2 field named tuples) in 80 chunks:
# works

julia> data[["AUD/JPY","USD/JPY"],:]
2-d Distributed NDSparse of (2 field named tuples) in 80 chunks:
# works

julia> data[["EUR/JPY","USD/JPY"], :]
Error showing value of type JuliaDB.DNDSparse{NamedTuples._NT_pair_timestamp{String,DateTime},NamedTuples._NT_bid_ask{Float32,Float32}}:
ERROR: On worker 2:
ArgumentError: reducing over an empty collection is not allowed
mr_empty_iter at ./reduce.jl:257
mapfoldl at ./reduce.jl:69
approx_size at /home/mbauman/.julia/v0.6/MemPool/src/MemPool.jl:85
approx_size at /home/mbauman/.julia/v0.6/MemPool/src/MemPool.jl:67 [inlined]
approx_size at /home/mbauman/.julia/v0.6/JuliaDB/src/util.jl:237
approx_size at /home/mbauman/.julia/v0.6/JuliaDB/src/util.jl:227
approx_size at /home/mbauman/.julia/v0.6/JuliaDB/src/util.jl:231
#poolset at ./<missing>:0
#tochunk#44 at /home/mbauman/.julia/v0.6/Dagger/src/chunks.jl:87
#tochunk at ./<missing>:0
do_task at /home/mbauman/.julia/v0.6/Dagger/src/scheduler.jl:255
#106 at ./distributed/process_messages.jl:268 [inlined]
run_work_thunk at ./distributed/process_messages.jl:56
macro expansion at ./distributed/process_messages.jl:268 [inlined]
#105 at ./event.jl:73
```

Feel free to ping me to get access to `load("bindata/full-2016/")`.

**Comment by shashi** [comment]
This should be fixed with MemPool v0.0.11... Are you on it?

**Comment by mbauman** [comment]
Aha, thanks! No, I had still been on  v0.0.10; upgrading did the trick.
